File: frontend/src/app/service/auth.service.ts
import { Injectable } from '@angular/core';
import {Observable} from "rxjs";
import {ApiResponse} from "../model/ApiResponse";
import {HttpClient} from "@angular/common/http";
import {SignUpDTO} from "../model/dto/user/SignUpDTO";
import {LoginDTO} from "../model/dto/user/LoginDTO";
import {TokenResponse} from "../model/TokenResponse";

@Injectable({
  providedIn: 'root'
})
export class AuthService {

  private url: string = `http://localhost:8091/api/auth`;

  constructor(private http: HttpClient) { }

  public signup(signUpDTO: SignUpDTO):Observable<ApiResponse> {
    return this.http.post<ApiResponse>(`${this.url}/sign-up`, signUpDTO);
  }

  public login(loginDTO: LoginDTO): Observable<TokenResponse> {
    return this.http.post<TokenResponse>(`${this.url}/sign-in`, loginDTO);
  }

}

File: fapi/src/main/java/by/kapitonov/cinema/fapi/service/AuthService.java
package by.kapitonov.cinema.fapi.service;

import by.kapitonov.cinema.fapi.rest.response.ApiResponse;
import by.kapitonov.cinema.fapi.rest.response.TokenResponse;
import by.kapitonov.cinema.fapi.service.dto.SignInDTO;
import by.kapitonov.cinema.fapi.service.dto.user.RegistrationUserDTO;

public interface AuthService {
    
    TokenResponse login(SignInDTO signInDTO);

    ApiResponse registration(RegistrationUserDTO registrationUserDTO);
    
}
